src/views/AddItem.jsx

```diff
@@ -1,3 +1,66 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+
+const defaultItem = { itemName: '', daysUntilNextPurchase: 7 };
+export function AddItem({ listToken }) {
+	const [item, setItem] = useState(defaultItem);
+	const [status, setStatus] = useState('');
+
+	const updateItem = (e) => {
+		if (status) setStatus('');
+		if (
+			e.target.name !== 'daysUntilNextPurchase' &&
+			e.target.name !== 'itemName'
+		)
+			return;
+		let updateVal = e.target.value;
+		if (e.target.name === 'daysUntilNextPurchase') {
+			updateVal = Number(updateVal);
+		}
+		setItem({ ...item, [e.target.name]: updateVal });
```

Comment:
nice job using spread operator to add things into the object, instead of change the object